inference-engine/src/transformations/src/transformations/common_optimizations/mvn_fusion.cpp

```diff
@@ -0,0 +1,339 @@
+        float eps_value;
+
+        if (!op::util::has_constant_value<float>(const_2_node, 2.0) || !op::util::get_single_value(const_eps_node, eps_value)) {
+            return false;
+        }
+
+        if (pattern_to_output.count(const_0_5) != 0) {
+            auto const_0_5_node = std::dynamic_pointer_cast<ngraph::opset6::Constant>(pattern_to_output.at(const_0_5).get_node_shared_ptr());
+
+            if (!const_0_5_node || !op::util::has_constant_value<float>(const_0_5_node, 0.5)) {
+                return false;
+            }
+        }
+
+        if (pattern_to_output.count(const_neg_1) != 0) {
+            auto const_neg_1_node = std::dynamic_pointer_cast<ngraph::opset6::Constant>(pattern_to_output.at(const_neg_1).get_node_shared_ptr());
+
+            if (!const_neg_1_node || !op::util::has_constant_value<float>(const_neg_1_node, -1)) {
+                return false;
+            }
+        }
+
+        auto axes_1_node = std::dynamic_pointer_cast<ngraph::opset6::Constant>(pattern_to_output.at(mean1_axes).get_node_shared_ptr());
+        auto axes_3_node = std::dynamic_pointer_cast<ngraph::opset6::Constant>(pattern_to_output.at(mean3_axes).get_node_shared_ptr());
+
+        if (!axes_1_node || !axes_3_node) {
+            return false;
+        }
+
+        auto axes_1_value = axes_1_node->cast_vector<int64_t>();
+        auto axes_3_value = axes_3_node->cast_vector<int64_t>();
+
+        if (!(axes_1_value == axes_3_value)) {
+            return false;
+        }
+        if (pattern_to_output.count(mean2_axes) != 0) {
+            auto axes_2_node = std::dynamic_pointer_cast<ngraph::opset6::Constant>(pattern_to_output.at(mean2_axes).get_node_shared_ptr());
+            if (!axes_2_node) {
+                return false;
+            }
+            auto axes_2_value = axes_2_node->cast_vector<int64_t>();
+            if (!(axes_1_value == axes_2_value)) {
+                return false;
+            }
+        }
+
+        auto mvn = std::make_shared<ngraph::opset6::MVN>(exp_input, axes_1_node, true, eps_value, op::MVNEpsMode::INSIDE_SQRT);
+
+        ngraph::NodeVector nodes_to_copy_info({ pattern_to_output.at(mean1).get_node_shared_ptr(),
+                                                pattern_to_output.at(sub1).get_node_shared_ptr(),
+                                                pattern_to_output.at(power).get_node_shared_ptr(),
+                                                pattern_to_output.at(mean3).get_node_shared_ptr(),
+                                                pattern_to_output.at(add_eps).get_node_shared_ptr() });
+
+        if (pattern_to_output.count(mean2) != 0 && pattern_to_output.count(sub2) != 0) {
+            nodes_to_copy_info.push_back(pattern_to_output.at(mean2).get_node_shared_ptr());
+            nodes_to_copy_info.push_back(pattern_to_output.at(sub2).get_node_shared_ptr());
+        }
+
+        if (pattern_to_output.count(cast) != 0) {
```

Comment:
And in all other places.
```suggestion
        if (pattern_to_output.count(cast)) {
```